tests/system/framework/roles/shadow.py

```diff
@@ -140,3 +144,126 @@ def chage(self, *args) -> ProcessResult:
         self.host.discard_file("/etc/shadow")
 
         return cmd
+
+    def newusers(self, *args, users_data: str = None) -> ProcessResult:
```

Comment:
I know the command is ```newusers``` I would consider renaming the method to something like ```user_batch_add```, I'm on the fence about this suggestion, it didn't make sense until I found out that the command is newusers, where there would be no mistaking what the method does.